src/main/scala/nl.knaw.dans.easy.deposit/docs/CollectionUtils.scala

```diff
@@ -31,7 +32,16 @@ object CollectionUtils {
   }
 
   implicit class RichOptionSeq[T](val sources: Option[Seq[T]]) extends AnyVal {
+
+    /** filters elements without a value (before mapping) */
     def withNonEmpty: FilterMonadic[T, Seq[T]] = sources.toSeq.flatten.withNonEmpty
+
+    /** filters empty XML elements after mapping (in case another field is mapped to a value) */
+    def toNonEmptyMap(f: T => Elem): Seq[Elem] = {
```

Comment:
1. There is only one use case for this. Are you planning on using this more often? Otherwise abstracting it into here may be a bit early.
2. I'm not happy with the name of this function `toNonEmptyMap` assumes you're returning a `Map` that has a type-level guarantee that it is not empty.
3. Why can't you use the usual `withNonEmpty.map(...)` instead in this single use case?